src/Client/Schedule/ScheduleHandle.php

```diff
@@ -0,0 +1,210 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Temporal\Client\Schedule;
+
+use ArrayIterator;
+use Countable;
+use DateTimeImmutable;
+use DateTimeInterface;
+use Google\Protobuf\Timestamp;
+use Temporal\Api\Schedule\V1\BackfillRequest;
+use Temporal\Api\Schedule\V1\SchedulePatch;
+use Temporal\Api\Schedule\V1\TriggerImmediatelyRequest;
+use Temporal\Api\Workflowservice\V1\DeleteScheduleRequest;
+use Temporal\Api\Workflowservice\V1\DescribeScheduleRequest;
+use Temporal\Api\Workflowservice\V1\ListScheduleMatchingTimesRequest;
+use Temporal\Api\Workflowservice\V1\PatchScheduleRequest;
+use Temporal\Api\Workflowservice\V1\UpdateScheduleRequest;
+use Temporal\Client\ClientOptions;
+use Temporal\Client\GRPC\ServiceClientInterface;
+use Temporal\Client\Schedule\Info\ScheduleDescription;
+use Temporal\Client\Schedule\Policy\ScheduleOverlapPolicy;
+use Temporal\Common\Uuid;
+use Temporal\DataConverter\DataConverterInterface;
+use Temporal\Exception\InvalidArgumentException;
+use Temporal\Internal\Mapper\ScheduleMapper;
+use Temporal\Internal\Marshaller\MarshallerInterface;
+use Temporal\Internal\Marshaller\ProtoToArrayConverter;
+use Traversable;
+
+final class ScheduleHandle
+{
+    public function __construct(
+        private readonly ServiceClientInterface $client,
+        private readonly ClientOptions $clientOptions,
+        private readonly DataConverterInterface $converter,
+        private readonly MarshallerInterface $marshaller,
+        private readonly ProtoToArrayConverter $protoConverter,
+        private readonly string $namespace,
+        private readonly string $id,
+    ) {
+    }
+
+    /**
+     * GetID returns the schedule ID associated with this handle.
+     */
+    public function getID(): string
+    {
+        return $this->id;
+    }
+
+    /**
+     * Update the Schedule.
+     *
+     * NOTE: If two Update calls are made in parallel to the same Schedule there is the potential
+     * for a race condition.
+     */
```

Comment:
Probably worth mentioning what the conflict token param is good for, and that it can be obtained from describe